### PR TITLE
Use QgsNetworkAccessManager instead of urllib for network requests

### PR DESCRIPTION
**Pull request by ltbam**
- Use of QNetworkManager instead of urllib for network access (proxy support).
- Fixed some UTF encoding issues
- Added some WMS types for link recognition.

### CODE REVIEW

**Comment by m-kuhn** [comment]
Assigned @tomkralidis

File: python/ext-libs/owslib/csw.py [hunk 1]
```diff
@@ -267,7 +267,15 @@ def getrecordbyid(self, id=[], esn='full', outputschema=namespaces['csw'], forma
             'id': ','.join(id),
         }
 
-        self.request = '%s%s' % (bind_url(self.url), urlencode(data))
+        l = []
+        data = data.items()
+        for k, v in data:
+            k = str(k)
```
Comment (m-kuhn) [inline python/ext-libs/owslib/csw.py]: `str` should not be used for strings.

 * Remove it if you are sure that there is a string returned always anyway
 * Change it to `unicode()` if you need to convert it to a string

**Comment by m-kuhn** [comment]
I think owslib is an external dependency. Patching it makes it harder to maintain it (i.e. if it's updated to a newer version the patch will need to be re-applied).

I would like to get @tomkralidis feedback on this.

-----------------

Alternatively, would it be possible to set [environment variables for proxies](https://docs.python.org/2/library/urllib.html) to get urllib working properly? This approach would fix all occurrences of urllib and leave the dependency code untouched.

**Comment by tomkralidis** [comment]
OWSLib is an external dependency that we will not patch here.  If there are OWSLib specific changes required, this is better realized by making them in OWSLib proper.  Note that the goal is for OWSLib to be completely upstream (and outside) of QGIS proper.

I would recommend this PR be closed and a new PR issued with only MetaSearch specific changes.

**Comment by m-kuhn** [comment]
@tomkralidis what do you think about the environment variables for urllib approach?
That should not need upstream adjustments as far as I can tell.

**Comment by tomkralidis** [comment]
@m-kuhn +1 for use of environment variables (`http_proxy`, I'm guessing).

**Comment by m-kuhn** [comment]
@ltbam could you implement these changes?

**Comment by m-kuhn** [comment]
@ltbam any feedback on this?

**Comment by m-kuhn** [comment]
Closed due to lack of feedback. Will be great if somebody will be able to implement this.

**Pull request closed without merging**